metadata-io/src/main/java/com/linkedin/metadata/graph/GraphService.java

```diff
@@ -93,6 +95,19 @@ RelatedEntitiesResult findRelatedEntities(@Nullable final String sourceType, @No
   @Nonnull
   default EntityLineageResult getLineage(@Nonnull Urn entityUrn, @Nonnull LineageDirection direction, int offset,
       int count, int maxHops) {
+    return getLineage(entityUrn, direction, new GraphFilters(new ArrayList(getLineageRegistry().getEntitiesWithLineage())), offset, count, maxHops);
+  }
+
+  /**
+   * Traverse from the entityUrn towards the input direction up to maxHops number of hops. If entityTypes is not empty,
+   * will only return edges to entities that are within the entity types set.
+   * Abstracts away the concept of relationship types
+   *
+   * Unless overridden, it uses the lineage registry to fetch valid edge types and queries for them
+   */
+  @Nonnull
+  default EntityLineageResult getLineage(@Nonnull Urn entityUrn, @Nonnull LineageDirection direction,
```

Comment:
Is there a way we can only filter for entity types that the current URN is related to via a lineage edge?

Reply:
It seems we are just filtering for anything that has lineage... That will solve our immedaite problem with data process instance but in the long tem could this pop up again?

Reply:
addressed- only relevant entity types will be pulled